backend/test/map/map.service.test.ts

```diff
@@ -0,0 +1,345 @@
+
+describe('MapService 테스트', () => {
+  let mapService: MapService;
+  let mapRepository: jest.Mocked<MapRepository>;
+  let userRepository: jest.Mocked<UserRepository>;
+  let placeRepository: jest.Mocked<PlaceRepository>;
+  let fakeUser1: User;
+  let page: number;
+  let pageSize: number;
+  beforeAll(() => {
+    fakeUser1 = {
+      id: 1,
+      ...UserFixture.createUser({ oauthId: 'abc' }),
+    };
+    [page, pageSize] = [1, 10];
+  });
+  beforeEach(async () => {
+    mapRepository = {
+      searchByTitleQuery: jest.fn(),
+      findAll: jest.fn(),
+      count: jest.fn(),
+      findById: jest.fn(),
+      findByUserId: jest.fn(),
+      save: jest.fn(),
+      softDelete: jest.fn(),
+      update: jest.fn(),
+      existById: jest.fn(),
+    } as unknown as jest.Mocked<MapRepository>;
+    userRepository = {
+      findByProviderAndOauthId: jest.fn(),
+      createUser: jest.fn(),
+      findById: jest.fn(),
+      existById: jest.fn(),
+    } as unknown as jest.Mocked<UserRepository>;
+    placeRepository = {
+      findByGooglePlaceId: jest.fn(),
+      findAll: jest.fn(),
+      searchByNameOrAddressQuery: jest.fn(),
+      existById: jest.fn(),
+    } as unknown as jest.Mocked<PlaceRepository>;
+    mapService = new MapService(mapRepository, userRepository, placeRepository);
+    userRepository.existById.mockResolvedValue(true);
+  });
+  describe('searchMap 메소드 테스트', () => {
+    it('파라미터 중 query 가 없을 경우 공개된 모든 지도를 반환한다.', async () => {
+      const mockMaps: Map[] = createPublicMaps(5, fakeUser1).map((map) => {
+        return {
+          ...map,
+          mapPlaces: [],
+        };
+      });
+      const spyFindAll = mapRepository.findAll.mockResolvedValue(mockMaps);
+      const spyCount = mapRepository.count.mockResolvedValue(mockMaps.length);
+
+      const result = await mapService.searchMap(undefined, 1, 10);
+      const expectedMaps = await Promise.all(
+        mockMaps.map(async (mockMap) => await MapListResponse.from(mockMap)),
+      );
+      expect(spyFindAll).toHaveBeenCalledWith(page, pageSize);
+      expect(result.maps).toEqual(
+        expect.arrayContaining(
+          expectedMaps.map((map) => expect.objectContaining(map)),
+        ),
+      );
+      expect(spyCount).toHaveBeenCalledWith({
+        where: { title: undefined, isPublic: true },
+      });
+      expect(result.currentPage).toEqual(page);
+      expect(result.totalPages).toEqual(Math.ceil(mockMaps.length / pageSize));
+    });
+    it('파라미터 중 쿼리(지도 title)가 있을 경우 해당 제목을 가진 지도들을 반환한다', async () => {
+      const searchTitle = 'cool';
+      const mockCoolMaps: Map[] = createPublicMapsWithTitle(
+        5,
+        fakeUser1,
+        'cool map',
+      ).map((map) => {
+        return {
+          ...map,
+          mapPlaces: [],
+        };
+      });
+      const spySearchByTitleQuery = jest
+        .spyOn(mapRepository, 'searchByTitleQuery')
+        .mockResolvedValue(mockCoolMaps);
```

Comment:
p1: 위에서 모킹했는데 왜 다시 `spyOn`을 사용하는걸까요?
```suggestion
      const spySearchByTitleQuery = mapRepository.searchByTitleQuery.mockResolvedValue(mockCoolMaps)
```